tests/backup/backup_test.go

```diff
@@ -6408,6 +6408,244 @@ var _ = Describe("{IssueMultipleDeletesForSharedBackup}", func() {
+				cloudCredUID = uuid.New()
+				CloudCredUIDMap[cloudCredUID] = cloudCredName
+				CreateCloudCredential(provider, cloudCredName, cloudCredUID, orgID)
+			}
+		})
+
+		Step("Register cluster for backup", func() {
+			ctx, _ := backup.GetAdminCtxFromSecret()
+			err := CreateSourceAndDestClusters(orgID, "", "", ctx)
+			dash.VerifyFatal(err, nil, "Creating source and destination cluster")
+			clusterStatus, clusterUid = Inst().Backup.RegisterBackupCluster(orgID, SourceClusterName, "")
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, "Verifying backup cluster")
+		})
+
+		Step("Creating backup location", func() {
+			log.InfoD("Creating backup location")
+			providers := getProviders()
+			for _, provider := range providers {
+				cloudCredName = fmt.Sprintf("%s-%s-%v", "cred", provider, timeStamp)
+				backupLocation = fmt.Sprintf("autogenerated-backup-location-%v", time.Now().Unix())
+				backupLocationUID = uuid.New()
+				backupLocationMap[backupLocationUID] = backupLocation
+				err := CreateBackupLocation(provider, backupLocation, backupLocationUID, cloudCredName, cloudCredUID,
+					getGlobalBucketName(provider), orgID, "")
+				dash.VerifyFatal(err, nil, "Creating backup location")
+			}
+		})
+
+		Step("Start backup of application to bucket", func() {
+			for _, namespace := range bkpNamespaces {
+				ctx, err := backup.GetAdminCtxFromSecret()
+				dash.VerifyFatal(err, nil, "Getting context")
+				backupName = fmt.Sprintf("%s-%s-%s", BackupNamePrefix, namespace, backupLocationName)
+				err = CreateBackup(backupName, SourceClusterName, backupLocation, backupLocationUID, []string{namespace},
+					nil, orgID, clusterUid, "", "", "", "", ctx)
+				log.FailOnError(err, "Backup creation failed for backup %s", backupName)
+				backupNames = append(backupNames, backupName)
+			}
+		})
+
+		Step("Create users", func() {
+			log.InfoD("Creating users")
+			users = createUsers(numberOfUsers)
+		})
+
+		Step("Share Backup with multiple users", func() {
+			log.InfoD("Sharing Backup with multiple users")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			dash.VerifyFatal(err, nil, "Getting context")
+			err = ShareBackup(backupName, nil, users, ViewOnlyAccess, ctx)
+			log.FailOnError(err, "Failed to share backup with users")
+		})
+
+		Step("Restart backup pod when backup sharing is in-progress", func() {
+			log.InfoD("Restart backup pod when backup sharing is in-progress")
+			backuPodLabel := make(map[string]string)
+			backuPodLabel["app"] = "px-backup"
+			backupPodNamespace := "px-backup"
+			pods, err := core.Instance().GetPods("px-backup", backuPodLabel)
+			dash.VerifyFatal(err, nil, "Geting px-backup pod")
+			log.Infof("Backup pods are %v", pods.Items)
+			for _, pod := range pods.Items {
+				err := core.Instance().DeletePod(pod.GetName(), backupPodNamespace, false)
+				log.FailOnError(err, fmt.Sprintf("Failed to kill pod [%s]", pod.GetName()))
+			}
+			time.Sleep(5 * time.Minute)
```

Comment:
Instead of using `time.Sleep` can we use `task.DoRetryWithTimeout` which is faster and more efficient?